### PR TITLE
Improved logging & a test for T3

### PR DESCRIPTION
**Pull request by rupertbates**
## What does this change?
This PR justs adds some additional logging to the soft-opt-in-consent-setter lambda as currently it is very minimal. It also adds a test for the new Tier Three product although I'm not sure there's a massive amount of value to it as it doesn't use the actual json file where consents are defined (this is in S3).

I think it would an improvement to move the definition of consents from this json file into this repo where it can be properly tested and version controlled, but that will be a separate PR.

### CODE REVIEW

File: handlers/soft-opt-in-consent-setter/src/main/scala/com/gu/soft_opt_in_consent_setter/IdentityConnector.scala [hunk 1]
```diff
@@ -1,7 +1,9 @@
 package com.gu.soft_opt_in_consent_setter
 
 import com.gu.soft_opt_in_consent_setter.models.{IdentityConfig, SoftOptInError}
+import com.typesafe.scalalogging.LazyLogging
```
Comment (johnduffell) [inline handlers/soft-opt-in-consent-setter/src/main/scala/com/gu/soft_opt_in_consent_setter/IdentityConnector.scala:4]: does this import get used?

Comment (rupertbates) [reply handlers/soft-opt-in-consent-setter/src/main/scala/com/gu/soft_opt_in_consent_setter/IdentityConnector.scala:4]: Nope good spot

File: handlers/soft-opt-in-consent-setter/src/test/scala/com.gu.soft_opt_in_consent_setter/HandlerTests.scala [hunk 2]
```diff
@@ -261,6 +261,53 @@ class HandlerTests extends AnyFunSuite with Matchers with MockFactory {
       mockSfConnector,
     )
 
+    result shouldBe Right(())
+  }
+  test(testName = "processAcquiredSub should handle a Tier Three acquisition event correctly") {
+    mockSendConsentsReq
+      .expects(
```
Comment (johnduffell) [inline handlers/soft-opt-in-consent-setter/src/test/scala/com.gu.soft_opt_in_consent_setter/HandlerTests.scala:268]: it's not quite clear when the assertion is actually made, but I assume it fails if the mockSendConsentsReq is not called?
I assume there's some tearDown logic called that will check all these at the end of the test.
I'm used to the mockito style where we do all the assertions afterwards that the mock is called.

Comment (rupertbates) [reply handlers/soft-opt-in-consent-setter/src/test/scala/com.gu.soft_opt_in_consent_setter/HandlerTests.scala:268]: Yes it is different to most of the tests we have but it seems to be the pattern in this project

File: handlers/soft-opt-in-consent-setter/src/test/scala/com.gu.soft_opt_in_consent_setter/testData/ConsentsCalculatorTestData.scala [hunk 3]
```diff
@@ -20,6 +27,7 @@ object ConsentsCalculatorTestData {
     "guardianweekly" -> guWeeklyMapping,
     "testproduct" -> testProductMapping,
     "FeastInAppPurchase" -> testFeastMobileSubscriptionMapping,
+    "Tier Three" -> testTierThreeMapping,
```
Comment (johnduffell) [inline handlers/soft-opt-in-consent-setter/src/test/scala/com.gu.soft_opt_in_consent_setter/testData/ConsentsCalculatorTestData.scala:30]: I know it's test data but why not be more consistent with the existing product names?  or are the existing ones inconsistent with PROD?

**Review by johnduffell — APPROVED** [review]
looks good although see comments, nothing show stopping though.

File: handlers/soft-opt-in-consent-setter/src/test/scala/com.gu.soft_opt_in_consent_setter/testData/ConsentsCalculatorTestData.scala (hunk 3 above)
Comment (rupertbates) [reply handlers/soft-opt-in-consent-setter/src/test/scala/com.gu.soft_opt_in_consent_setter/testData/ConsentsCalculatorTestData.scala:30]: The existing ones are inconsistent with what is in PROD, they should map to the product names in Salesforce.

**Pull request merged**